torchvision/transforms/functional_tensor.py

```diff
@@ -619,6 +619,22 @@ def resize(img: Tensor, size: List[int], interpolation: int = 2) -> Tensor:
     return img
 
 
+def _gen_affine_grid(
+        theta: Tensor, w: int, h: int, ow: int, oh: int,
+) -> Tensor:
+    d = 0.5
+    x = torch.arange(ow) + d - ow * 0.5
+    y = torch.arange(oh) + d - oh * 0.5
+
+    y, x = torch.meshgrid(y, x)
+    pts = torch.stack([x, y, torch.ones_like(x)], dim=-1)
+    output_grid = torch.matmul(pts, theta.t())
+
+    output_grid = output_grid / torch.tensor([0.5 * w, 0.5 * h])
```

Comment:
Here is the principal difference to `affine_grid`-like implementation. In `affine_grid`-lik implementation it would be
```python
    x = (torch.arange(ow) + d - ow * 0.5) / (0.5 * w)
    y = (torch.arange(oh) + d - oh * 0.5) / (0.5 * h)
```
instead of `output_grid` scaling.